src/fhirtypes/common.ts

```diff
@@ -980,6 +980,34 @@ export function replaceReferences<T extends AssignmentRule | CaretValueRule>(
         (clone.value as FshReference).reference = value.reference.slice(0, firstPipe);
         clone = replaceReferences(clone, tank, fisher);
       }
+      // if the reference to an entity is provided but is unable to be resolved
+      else if (type == null && id == null && value.reference) {
+        // if reference does not have type / id reference format
+        // example: Patient-oops
+        if (!value.reference.includes('/')) {
+          logger.warn(
+            `Cannot find the entity referenced at ${value.reference}. The provided reference value will be used, however, this reference does not conform to the FHIR Reference() format.`
+          );
+        }
+
+        // if reference is a relative reference with too many parts
+        // example: PatientExample/Patient/Patient-oops
+        if (
+          value.reference.includes('/') &&
+          value.reference.indexOf('/') !== value.reference.lastIndexOf('/')
+        ) {
```

Comment:
Based on an example in the FHIR documentation, I think that a value with more than one `/` might actually be acceptable? The last example in the [Literal References](https://hl7.org/fhir/R5/references.html#literal) section shows `Observation/1x2/_history/2` as a version-specific reference value. So maybe, only a value without _any_ `/` would be one that we warn about? I'd like to get opinions from the rest of the team on this.